clj-refactor.el

```diff
@@ -1983,37 +1983,68 @@ the alias in the project."
     (backward-sexp 1)
     (looking-at-p "[-+0-9]")))
 
+(defun completable-for-cljr-slash? (sym)
+  (when sym
+    (not (null (string-match-p "^\\([a-zA-Z]+[a-zA-Z0-9\\-]*\\)+\\(\\.?[a-zA-Z]+[a-zA-Z0-9\\-]*\\)*$" sym)))))
+
+(progn
```

Comment:
Quite obviously I don't intend to ship `assert`s, but seemingly there weren't unit tests in the test suite.

I do appreciate the `features` but I'm unfamiliar with their Emacs variant. More importantly this seems the perfect thing that can be tested directly in terms of inputs->outputs for a specific, pure `defun`.

If unit tests SGTY, I'd appreciate some guidance on how to actually ship them into CI.

Reply:
I've used ERT in the past, but CIDER uses [buttercup](https://github.com/jorgenschaefer/emacs-buttercup) which addresses some of the shortcomings of ERT.

If you're up for it, it would be great to use the same for clj-refactor. If so, we could probably just copy the setup that's in use in CIDER. 

Does that sound like a good idea @bbatsov or do you have some reservations about `buttercup` after using it in CIDER?

@vemv the current test suit in clj-refactor is written in [ecukes](https://github.com/ecukes/ecukes) and we've been wanting to get away from that for some time, since it stopped being maintained.  As I look up the link now I see that it's been moved to its own org and seems to have acquired a new maintainer.  Perhaps there's some hope in getting our test suite back into green...

I have mixed feelings about ecukes.  In general it's a really good fit, but in practice it has been quite finicky to work with.  It would be a lot of work to rewrite the features, though, so if we can update our ecukes version and get them running again that would probably be the best solution.

Reply:
Buttercup is awesome. No reservations about it at all.